### ISSUE TITLE
Imports of dependencies of dependencies not found `vitest 

### ISSUE THREAD

**Issue by silverbucket** [vitest-dev/vitest#2501]
### Describe the bug

After upgrading from `0.25.5` to any more recent version up to the current `0.25.8` I'm getting import errors of the dependencies of those imported packages. This isn't happening with all of them, but for example it is with Sentry.

```
        "@sentry/svelte": "^7.26.0",
        "@sentry/tracing": "^7.26.0",
```

Works fine during runtime, however when running tests, I get the message:

```
Error: Failed to load url @sentry/browser (resolved id: @sentry/browser). Does the file exist?
```

This is only silenced by manually adding `@sentry/browser` as a direct dependency, however then the next messages is about `@sentry/util` and so on.

### Reproduction

Add `sentry/browser` as a dependency and try to write a test with it.

### System Info

```shell
System:
    OS: macOS 12.6.1
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 145.65 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.18.0 - ~/.nvm/versions/node/v16.18.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.18.0/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.18.0/bin/npm
  Browsers:
    Brave Browser: 108.1.46.134
    Chrome: 108.0.5359.98
    Firefox: 105.0.3
    Safari: 16.1
  npmPackages:
    vite: ^4.0.1 => 4.0.1
    vitest: 0.25.5 => 0.25.5
```
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by binarious** [comment]
Similar issue here. 0.25.5 works and 0.25.6+ doesn't find some sub-dependencies. In my case a project using [svelte-i18n](https://github.com/kaisermann/svelte-i18n) results in the following error:
> Error: Failed to load url deepmerge (resolved id: deepmerge). Does the file exist?

**Comment by binarious** [comment]
Reproduction: https://stackblitz.com/edit/vitest-dev-vitest-cfjmfi?file=test/basic.test.ts

**Comment by hagai-reuveni** [comment]
Hey,
I'm getting the same error when running v.0.28.3. (vite 4.0.4, pnpm).
Does anyone experience the same issue in current versions?

**Comment by sheremet-va** [comment]
> Hey,
> I'm getting the same errors when running v.0.28.3. (vite 4.0.4, pnpm). With both `sentry` and `@felte`.
> Does anyone experience the same issue in current versions?

Yes, this will be fixed by #2777

**Comment by markteekens** [comment]
` Failed to load url ~~/store/pos-information (resolved id: ~~/store/pos-information). Does the file exist?`

This has something to do with `~~/`, is there a way to include this?
